### PR TITLE
API for dynamic text displays in VCV-ported modules, and using them in NoisePlethora

### PR DESCRIPTION
**Pull request by hemmer**
discussion here: https://forum.4ms.info/t/pattern-for-using-dynamictextdisplay-with-vcv-based-mm-plugins/317

https://github.com/user-attachments/assets/c896069a-1572-4cd9-82f2-061746c48181

### CODE REVIEW

File: firmware/vcv_plugin/export/src/module_widget_adaptor.hh [hunk 1]
```diff
@@ -140,6 +140,20 @@ struct ModuleWidgetAdaptor {
 			pr_err("Error: can't add a null SvgWidget\n");
 	}
 
+	void addText(rack::ui::TextField *widget) {
+		if (widget) {
+			widget->element = make_element(widget);
+			assign_element_fields(widget, "");
+
+			ElementCount::Indices indices = clear();
+			indices.light_idx = 0; // widget->firstLightId;
```
Comment (hemmer) [inline firmware/vcv_plugin/export/src/module_widget_adaptor.hh:149]: What should this be? I might want to add multiple LCD displays. I don't have a LIGHT in the VCV sense assigned for these displays, should i?

File: firmware/vcv_plugin/export/src/module_widget_adaptor.hh [hunk 2]
```diff
@@ -156,6 +170,7 @@ struct ModuleWidgetAdaptor {
 				el.coords = Coords::TopLeft;
 				el.short_name = name;
 				el.long_name = name;
+				el.visible = widget->visible;
```
Comment (hemmer) [inline firmware/vcv_plugin/export/src/module_widget_adaptor.hh:173]: Ah forgot to remove, but actually would you be willing to add `bool visible` to BaseElement? Currently visibility status not respected (useful for hidden params).

File: firmware/vcv_plugin/internal/make_element.cc [hunk 3]
```diff
@@ -486,6 +486,14 @@ Element make_element(rack::app::SvgButton *widget) {
 	return make_element(widget->sw);
 }
 
+Element make_element(rack::ui::TextField *widget) {
+	DynamicTextDisplay element{};
+	element.text = widget->text;
+	element.font = "Segment32";
+	element.color = Colors565::Red;
```
Comment (hemmer) [inline firmware/vcv_plugin/internal/make_element.cc:493]: looks like this: 

https://github.com/user-attachments/assets/98dc38a6-6e06-44b7-a555-f769384e75f2

**Comment by danngreen** [comment]
Is working well! Also the changes to the rack-interface/plugin-sdk are enough to let third-party plugins take advantage of this feature.

To test the API I easily converted the CountModule VC Pulse Divider module, and it works as expected.

This will bump the SDK API version up, but it's backwards compatible (only adds symbols, does not remove or change meaning of existing symbols), so the next firmware release will need to be v1.3

I also did test for backwards compatibility (using older plugins on this branch), and it works fine.
And I tested for forward compatibility (using plugins compiled with this plugin-sdk version, loading with the fw v1.2.1) and it works as long as plugins don't try to use the VCVTextDisplay (obviously that won't work).

This does bring up the issue of 3rd party fonts... I'll open an issue for that.

**Pull request merged**